### PR TITLE
HPCC-16381 Add GetNumFileToCopy

### PR DESCRIPTION
**Pull request by miguelvazq**
Add deploy complete notification on the cluster level for Roxie clusters.

Signed-off-by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please comment

**Comment by GordonSmith** [comment]
@miguelvazq Is this for 6.4?

**Comment by miguelvazq** [comment]
@GordonSmith Good catch will rebase

**Comment by miguelvazq** [comment]
@GordonSmith Rebased.

File: esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            if (this.inherited(arguments))
+                return;
+            this.cluster = params.__hpcc_treeItem.Name;
+
+            this._refreshActionState();
+            this.refreshGrid();
+        },
+
+        postCreate: function (args) {
+            var context = this;
```
Comment (GordonSmith) [inline esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js]: Not needed

File: esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            this.inherited(arguments);
+        },
```
Comment (GordonSmith) [inline esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js]: The whole postCreate is not needed.

File: esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        createGrid: function (domID) {
+            var context = this;
```
Comment (GordonSmith) [inline esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js]: Not used.

File: esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            var retVal = new declare([ESPUtil.Grid(true, true)])({
+                store: this.store,
+                columns: {
+                    URL: {label: this.i18n.URL, width:100, sortable: false},
+                    Status: {label: this.i18n.Status, width: 30, sortable: false},
+                    NumQueryFileToCopy: {label:this.i18n.FilesPending, width: 30, sortable: false}
```
Comment (GordonSmith) [inline esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js]: One column should not have a width set?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith Updated.

**Comment by miguelvazq** [comment]
@GordonSmith Updated.

**Comment by miguelvazq** [comment]
@rpastrana Please review translations.

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 5]
```diff
@@ -206,6 +206,7 @@ define(
     FilePath: "Camino del archivo",
     FilePermission: "Permisos de Archivos",
     Files: "Archivos",
+    FilesPending: "Archivos pendiente",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/es/hpcc.js]: I think we need to pluralize the "pendiente" as well

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 6]
```diff
@@ -533,6 +534,7 @@ define(
     RetainSuperfileStructure: "Retener estructura del super-archivo",
     RetypePassword: "Confirme la contraseña",
     Reverse: "Marchar Atrás",
+    RoxieFileCopy: "Estado de copia para archivos de Roxie",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/es/hpcc.js]: "de" instead of "para"

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 7]
```diff
@@ -647,6 +649,7 @@ define(
     title_ECLPlayground: "ECL Playground",
     title_ErrorsWarnings: "Errores/Advertencias de",
     title_EventScheduleWorkunit: "Planificador de Eventos",
+    title_FilesPendingCopy: "Archivos pendiente de copia",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/es/hpcc.js]: Archivos pendiente para copiar

**Review by rpastrana — COMMENTED** [review]
@miguelvazq couple of comments

File: esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
+    "dojo/i18n",
+    "dojo/i18n!./nls/hpcc",
+    "dojo/_base/array",
+
+    "hpcc/GridDetailsWidget",
+
+    "hpcc/DelayLoadWidget",
```
Comment (GordonSmith) [inline esp/src/eclwatch/GetNumberOfFilesToCopyWidget.js]: Not used

File: esp/src/eclwatch/TopologyWidget.js [hunk 9]
```diff
@@ -37,6 +37,7 @@ define([
     "hpcc/GridDetailsWidget",
     "hpcc/PreflightDetailsWidget",
     "hpcc/RequestInformationWidget",
+    "hpcc/GetNumberOfFilesToCopyWidget",
```
Comment (GordonSmith) [inline esp/src/eclwatch/TopologyWidget.js]: Is this used?  I think it is supposed to be in TopologyDetailsWidget?

File: esp/src/eclwatch/templates/TopologyDetailsWidget.html [hunk 10]
```diff
@@ -17,6 +17,8 @@
             </div>
             <div id="${id}_RequestInformation" title="${i18n.Preflight}" data-dojo-type="RequestInformationWidget">
             </div>
+            <div id="${id}_GetNumberOfFilesToCopy" title="${i18n.RoxieFileCopy}" data-dojo-type="GetNumberOfFilesToCopyWidget">
```
Comment (GordonSmith) [inline esp/src/eclwatch/templates/TopologyDetailsWidget.html]: "GetNumberOfFilesToCopyWidget" should be added to "TopologyDetailsWidget.js"

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by richardkchapman** [comment]
@miguelvazq

**Comment by miguelvazq** [comment]
@GordonSmith Updated per your comments

**Comment by miguelvazq** [comment]
@GordonSmith bump

**Review by GordonSmith — APPROVED** [review]
Looks fine.

**Pull request merged**